### ISSUE TITLE
--convert-3to4 converts common simple English words in comments such as: "on" -> "checked", "off" -> "unchecked"

### ISSUE THREAD

**Issue by hsandt** [godotengine/godot#73505]
### Godot version

v4.0.rc2.official [d2699dc7a]

### System information

Linux Ubuntu 20.04 with Unity desktop

### Issue description

I noticed that godot4 --convert-3to4 also converts names in comments, which is understandable when dealing with symbol names that are obviously Godot class/member names like `get_surface_material`. For instance, we could have a comment saying:

    # We use get_surface_material here

which would become

    # We use get_surface_override_material here

The problem is that it will convert all names in https://github.com/godotengine/godot/blob/master/editor/renames_map_3_to_4.cpp including simple common names, resulting in unwanted comments:

From demo platformer > Player.gd

\# If you define _physics_process on a node, Godot will call it every frame.

becomes

\# If you define _physics_process **checked** a node, Godot will call it every frame.

This won't apply mid-word, but will apply on words bounded by hyphens such as "on-again-off-again" -> "checked-again-unchecked-again".

Unfortunately, I see no easy solution for this: either
a. we run the risk to convert more strings and let the user revert bad conversions, or
b. we convert fewer strings and run the risk to miss a few ones such as `# we set the toggle to on here`

If we want to apply b, we need to distinguish common English words from others... We cannot remove entries from https://github.com/godotengine/godot/blob/master/editor/renames_map_3_to_4.cpp since we need them for true code conversion (not comment). So we'd need to add an extra table listing common English words so that we ignore them, for comments only.

Personally, I think that the user should always version control the project and monitor each change done by the converter. So we could just keep doing a. but warn the user that they should check for bad changes, at least if we auto-converted some words in comments (esp. not compounded words i.e. not containing an underscore).

If you go the b. way, here is the list of simple words I have found (may not be exhaustive):
on -> checked
off -> unchecked
raise -> move_to_top
clamped -> clamp
invert -> reverse
and i'm pretty sure no one says "stepify" ;)

Also note that 'exported' and 'tool' are _not_ converted in comments, only as symbols, taking the `@` prefix.

### Steps to reproduce

1. Run godot4 --convert-3to4 in the root folder of the demo project attached which contains a list of common words
2. Check the git diff (project is already version controlled)

Note: the project is already under Godot 4, not Godot 3, but it doesn't matter for text conversion.
If you want to create your own list of words, just write comments with the cimple words mentioned above (or find simple words from https://github.com/godotengine/godot/blob/master/editor/renames_map_3_to_4.cpp yourself)

Note 2: I found the list of common words in https://github.com/powerlanguage/word-lists/blob/master/1000-most-common-words.txt

### Minimal reproduction project

[Test convert3to4.zip](https://github.com/godotengine/godot/files/10768586/Test.convert3to4.zip)

**Comment by mbrealm** [comment]
Thanks @hsandt for raising this issue!

I recently ran a conversion from 3.5 to 4.0-rc2 and ran into the same problem (mainly with "on" -> "checked" - I didn't actually notice any of the others, maybe I just got lucky or maybe I missed them during inspection of the changes).

I would prefer if the converter NEVER did these replacements in code comments.
It is very easy to update comments correctly if the converter handles the code conversion, as one can and should step through the changes made and will notice if any comments around it are outdated.
